### ISSUE TITLE
Technical walkthroughs of FedRAMP Automation Tools

### ISSUE THREAD

**Issue by aj-stein-gsa** [GSA/fedramp-automation#653]
### This is a ...

improvement - something could be better

### This relates to ...

- [ ] the **FedRAMP OSCAL Registry**
- [ ] the **FedRAMP OSCAL baselines**
- [ ] the **Guide to OSCAL-based FedRAMP Content**
- [ ] the **Guide to OSCAL-based FedRAMP System Security Plans (SSP)**
- [ ] the **Guide to OSCAL-based FedRAMP Security Assessment Plans (SAP)**
- [ ] the **Guide to OSCAL-based FedRAMP Security Assessment Results (SAR)**
- [ ] the **Guide to OSCAL-based FedRAMP Plan of Action and Milestones (POA&M)**
- [ ] the **FedRAMP SSP OSCAL Template** (JSON or XML Format)
- [ ] the **FedRAMP SAP OSCAL Template** (JSON or XML Format)
- [ ] the **FedRAMP SAR OSCAL Template** (JSON or XML Format)
- [ ] the **FedRAMP POA&M OSCAL Template** (JSON or XML Format)

### User Story

As a FedRAMP stakeholder that will make or use tools to create OSCAL content for FedRAMP, I would like technical walkthroughs of the current tools they offer that I can integrate in my own work environment.

### Goals

- [ ] Documentation that explains
  - [ ] where and how you can obtain FedRAMP tools, data, and documentation
  - [ ] how to install and configure the tools, data, and documentation for my kind of environment
  - [ ] what FedRAMP tools are for and why
  - [ ] what FedRAMP data you can use with these tools and why
  - [ ] how I can use these tools with my OSCAL documents, adapt the documents, and validate my changes
  - [ ] where and how I can ask questions from FedRAMP maintainers and other community members for help when I am stuck

### Acceptance Criteria

- All FedRAMP Documents Related to OSCAL Adoption (https://github.com/GSA/fedramp-automation) affected by the changes in this issue have been updated.
- A Pull Request (PR) is submitted that fully addresses the goals of this User Story. This issue is referenced in the PR.

**Comment by aj-stein-gsa** [comment]
@kyhu65867, @Rene2mt, and I reviewed the feedback and will use that to drive a game plan and next steps I will document here and breakout task issues over this evening and tomorrow morning.

/cc @david-waltermire for awareness

**Comment by sam-aydlette** [comment]
@aj-stein-gsa this should include consideration for using the tools within a FedRAMP authorized environment (i.e what controls are covered, any customer compliance responsibilities)

**Comment by aj-stein-gsa** [comment]
> @aj-stein-gsa this should include consideration for using the tools within a FedRAMP authorized environment (i.e what controls are covered, any customer compliance responsibilities)

I love this feedback! This request is actually much in line with previous "dogfooding" strategies on building OSCAL components for the CLI tooling elsewhere. Combining the various components could also be interesting because of the modular nature of the tooling pieces (see the associated branch with PR 638; it [has a diagram for developers](https://github.com/aj-stein-gsa/fedramp-automation/blob/e4623ad6f4725525937b018a6e965d1b45bcd05e/src/validations/constraints/CONTRIBUTING.md#overview-of-the-constraint-and-testing-infrastructure#overview-of-the-constraint-and-testing-infrastructure) to understand the various pieces, of which only the constraint and testing files are actually in "this code base"). I also think the automation team should have a deeper think about this because I am not sure how much of the "developer tooling" would live in authorized environment (there is a lot of detail there and what we consider "staff workstations" but I would be very ... surprised ... if production automation systems used the full CLI and our test harness and not the underlying liboscal-java to build much less flexible REST services on purpose).

That said, let me jot down some notes for an issue to add this consideration as part of the epic on building better experience for the walkthrough. Before only Dave and I had mused to each other about this, I love seeing the larger staff rightfully ask about this. Thanks for commenting here!

**Comment by sam-aydlette** [comment]
> > @aj-stein-gsa this should include consideration for using the tools within a FedRAMP authorized environment (i.e what controls are covered, any customer compliance responsibilities)
> 
> I love this feedback! This request is actually much in line with previous "dogfooding" strategies on building OSCAL components for the CLI tooling elsewhere. Combining the various components could also be interesting because of the modular nature of the tooling pieces (see the associated branch with PR 638; it [has a diagram for developers](https://github.com/aj-stein-gsa/fedramp-automation/blob/e4623ad6f4725525937b018a6e965d1b45bcd05e/src/validations/constraints/CONTRIBUTING.md#overview-of-the-constraint-and-testing-infrastructure#overview-of-the-constraint-and-testing-infrastructure) to understand the various pieces, of which only the constraint and testing files are actually in "this code base"). I also think the automation team should have a deeper think about this because I am not sure how much of the "developer tooling" would live in authorized environment (there is a lot of detail there and what we consider "staff workstations" but I would be very ... surprised ... if production automation systems used the full CLI and our test harness and not the underlying liboscal-java to build much less flexible REST services on purpose).
> 
> That said, let me jot down some notes for an issue to add this consideration as part of the epic on building better experience for the walkthrough. Before only Dave and I had mused to each other about this, I love seeing the larger staff rightfully ask about this. Thanks for commenting here!

Of course!  Thanks for responding so thoughtfully.  I’m thinking about the overall industry use case, and it definitely makes sense to address general NIST 800-53/171 compliance as DOD IL5, SSDF, CMMC may apply to dev environments that support FedRAMP offerings.

**Comment by aj-stein-gsa** [comment]
@david-waltermire and @Rene2mt I have done much of the epic updates and we can build out from our discussion earlier today.
